das/daser.go

```diff
@@ -40,9 +40,9 @@ type DASer struct {
 func NewDASer(
 	da share.Availability,
 	hsub header.Subscriber,
-	getter header.Getter,
-	cstore datastore.Datastore,
-	bcast fraud.Broadcaster,
+	getter header.Store,
+	cstore datastore.Batching,
+	bcast fraud.Service,
```

Comment:
There was a conversation about this, and we decided to rip out the "proxy"-constructor for this instead. @renaynay any thoughts?

Reply:
I agree with you though, I think there should be a constructor that casts the types as before

Reply:
Let's go back and use the proxy constructor - it is unfortunately cleaner. And yes I agree we should offload ugliness on DI side rather than in actual constructors

Reply:
done